feature-libs/product/multi-dimensional/core/services/product-multi-dimensional.service.ts

```diff
@@ -74,29 +75,88 @@ export class ProductMultiDimensionalService {
         }
       });
 
-      variantCategories.push(variantCategory);
+      variantCategories.push(VariantCategoryGroup);
     });
 
     return this.checkIfEveryOptionHasImages(variantCategories);
   }
 
   /**
    * Checks if every variant option in the category has images.
+   *
+   * @param {VariantCategoryGroup[]} variantCategories - An array of variant categories.
+   * @returns {VariantCategoryGroup[]} - An array of variant categories with updated image information.
    */
   protected checkIfEveryOptionHasImages(
-    variantCategories: VariantCategory[]
-  ): VariantCategory[] {
-    return variantCategories.map((variantCategory: VariantCategory) => {
-      const variantOptions = variantCategory.variantOptions;
-      const hasImages =
-        variantCategory.hasImages ??
-        variantOptions.every((option) => option.images.length);
-
-      return {
-        ...variantCategory,
-        variantOptions,
-        hasImages,
-      };
-    });
+    variantCategories: VariantCategoryGroup[]
+  ): VariantCategoryGroup[] {
+    return variantCategories.map(
+      (VariantCategoryGroup: VariantCategoryGroup) => {
+        const variantOptions = VariantCategoryGroup.variantOptions;
+        const hasImages =
+          VariantCategoryGroup.hasImages ??
+          variantOptions.every((option) => option.images.length);
```

Comment:
To my understanding this implementation is not in line with your intentions described in the comment https://github.com/SAP/spartacus/pull/19040#discussion_r1687712623

I thought you want to set `hasImages` with `&&` (when both conditions are true), but not `??` ?:
```suggestion
        const hasImages =
          VariantCategoryGroup.hasImages &&
          variantOptions.every((option) => option.images.length);
```